examples/example-gateway/endpoints/contacts/save_contacts_test.go

```diff
@@ -0,0 +1,41 @@
+package contacts_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	clientContacts "github.com/uber/zanzibar/examples/example-gateway/build/gen-code/clients/contacts/contacts"
+	endpointContacts "github.com/uber/zanzibar/examples/example-gateway/build/gen-code/endpoints/contacts/contacts"
+	"github.com/uber/zanzibar/examples/example-gateway/build/services/example-gateway"
+)
+
+func TestSaveContactsCall(t *testing.T) {
+	ms := examplegatewayServiceGenerated.MustCreateTestService()
+	ms.Start()
+	defer ms.Stop()
+
+	endpointReqeust := &endpointContacts.SaveContactsRequest{
+		Contacts: []*endpointContacts.Contact{},
+	}
+	rawBody, _ := endpointReqeust.MarshalJSON()
+
+	clientRequest := &clientContacts.SaveContactsRequest{
+		UserUUID: "foo",
+	}
+	clientResponse := &clientContacts.SaveContactsResponse{}
+
+	ms.MockClientNodes().Contacts.On("SaveContacts", mock.Anything, mock.Anything, clientRequest).
```

Comment:
While it's fine for now, the interfaces here are not very discoverable. It would be better if we wrapped `mockery`-generated mocks (whose interface is `On(methodName string, arguments ...interface{})` and `Return(returnArguments ...interface{})`) with the concrete types (in this example, `OnSaveContacts(context.Context,
		reqHeaders map[string]string,
		args *clientsContactsContacts.SaveContactsRequest)` and `Return(*SaveContactsResponse, error)`).

Reply:
That's true, the mockery `On` API is very raw and flexible. Concretizing the mock client API will be a next step in the scope of fixture work.